### PR TITLE
Clean up dependency injection and configuration

### PR DESCRIPTION
**Pull request by zmbush**
Now add Jekyll::Assets.configure blocks

### CODE REVIEW

**Comment by zmbush** [comment]
This should resolve #134

**Comment by ixti** [comment]
IMO it worth to move all `lib/jekyll-assets/*.rb` under `lib/jekyll/assets/`:

```
./lib/jekyll/assets/bourbon.rb
./lib/jekyll/assets/compass.rb
...
```

And they should not be auto-required with `jekyll/assets`. So that instead of asking users to call `Jekyll::Assets::Bourbon.bind` we will ask them to add `require "jekyll/assets/bourbon"`

Inside such integration there will be a simple code:

``` ruby
require "jekyll/assets"
require "bourbon"

Jekyll::Assets.configure do |assets|
  root = Gem::Specification.find_by_name("bourbon").gem_dir
  assets.append_path File.join(root, "app", "assets", "stylesheets")
end
```

At some point I would like to move all those integrations into separate gems completely, e.g. `jekyll-assets-bourbon`, etc.

**Comment by zmbush** [comment]
The primary reason I did it the way I did was that was to make testing easier. For example, neat depends on bourbon, and since the 'config' is cleaned up before every test, then trying to 'require' bourbon again wouldn't work.

**Comment by ixti** [comment]
Well, probably it's time to start extracting those integrations into separate gems then? :D

**Comment by ixti** [comment]
As this all going to be released as part of `1.0.0` I believe we can break API as much as we want :D

**Comment by ixti** [comment]
After all I really want to get rid of all this integrations out of the core...

**Comment by zmbush** [comment]
These libraries are going to be factored out before we release 1.0.0, so maybe we could just merge this change now, and then we can start working on breaking the integrations into their own gems.

File: lib/jekyll-assets.rb [hunk 1]
```diff
@@ -1,2 +1,6 @@
 # rubocop: disable FileName
 require "jekyll/assets"
+
+Dir[File.dirname(__FILE__) + "/jekyll-assets/*.rb"].each do |file|
+  require file
+end
```
Comment (ixti) [inline lib/jekyll-assets.rb]: This should go to `lib/jekyll/assets`

**Comment by ixti** [comment]
Agree!

**Pull request merged**